Migrate/tutorials

returnMarcco opened the pull request:
# Description

- These changes migrate the `tutorials` component from `Coffeescript` and `AngularJS` to `Typescript` and `Angular`.

Fixes # (issue)

https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.planner/tt.c_19:bd20175d09414f079490a2403f7fca74@thread.tacv2_p_njykIFLDn0iAY1at7tACfcgADgBS_h_1714385623719?tenantId=d02378ec-1688-46d5-8540-1c28b5f470f6&webUrl=https%3A%2F%2Ftasks.teams.microsoft.com%2Fteamsui%2FpersonalApp%2Falltasklists&context=%7B%22subEntityId%22%3A%22%2Fv1%2Fplan%2FnjykIFLDn0iAY1at7tACfcgADgBS%2Ftask%2FOdDAKjGhdUOwm67HLnkFV8gANPFC%22%2C%22channelId%22%3A%2219%3Abd20175d09414f079490a2403f7fca74%40thread.tacv2%22%7D

## Type of change

- [x] Migration
- [x] This change requires a documentation update

# How Has This Been Tested?

Note: If you are reviewing this PR, you can also follow the below steps to verify functionality.
- Selected a unit from the dropdown at the top of the screen
- Chose the `task` dropdown beside the dropdown just mentioned, and chose `Tutorials`
- Verified that `before filtering by campus_id`, tutorials that were at a different campus would fail the auth check
- Implemented `filtering` logic and verified the list of tutorials were authorised for the `campus_id`  `student_1` belonged to
- Enrolled in a tutorial
- Withdrew from a tutorial

# Screenshots
Before
![before](https://github.com/user-attachments/assets/e0d4b0c3-2dd9-422f-bd82-9d31e307fff3)

After
![after](https://github.com/user-attachments/assets/99307d3a-89ae-48ae-a3ac-2e2ae6086517)

## Testing Checklist:

- [x] Tested in latest Chrome
- [ ] Tested in latest Safari (no access to a Mac for educational purposes)
- [x] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have requested a review from @macite and @jakerenzella on the Pull Request

zara1504 committed:
chore: remove media-service.coffee and modify doubtfire-angularjs.module.ts

macite committed:
Merge branch 'migrate/confirmation-modal' of https://github.com/b0ink/doubtfire-web into b0ink-migrate/confirmation-modal

returnMarcco committed:
chore: migrate tutorials

- unlink old component
- link new component
- delete old files
- add UI-router state declaration
- remove reference to old state

returnMarcco committed:
chore: migrate tutorials

- delete old files

returnMarcco committed:
chore: migrate tutorials

- declare and define `tutorials` Typescript class
- define `tutorials` markup in Angular and Angular material
- add stylesheet

returnMarcco committed:
chore: migrate tutorials

- delete old template

zara1504 commented:
Hey I just tested out you're migration. I didn't get any errors and your code looks good. Following the testing above the component matches the images you have provided, and the withdraw/enroll button works well with, testing it on a unit that had 2 tutorials it would only allow you to enroll into one. Good job :)

sodabottlebaby commented:
Hi Jason! Thank you so much for your help throughout this trimester and great work on your PR! 

I was able to successfully pull your PR and run it. I didn't get any errors except for the animation warning which isn't related to your code. The component matches the after migration image, the buttons are working properly and your actual code is well-written and well-documented so I understood each section properly.

I approve the PR and thank you once again! :)

martindolores reviewed (approved):
Looks like there were no errors or warnings introduces, and your code looks solid! After testing, the component matches images provided, and the everything looks like its working as intended. Nice work 🤘

returnMarcco commented:
Thanks for the reviews everyone

HasinduWelarathne commented:
Hey,
I looked through all five files changed works end-to-end. No new errors or warnings have appeared. All functionality functions correctly. Good work!

aNebula commented on src/app/projects/states/tutorials/tutorials.component.tpl.html:
Please take `.tpl` out from the filename. Here's migration convention:
![image](https://github.com/user-attachments/assets/76fabf1c-7cfe-4c6d-9239-b73190f7e64d)

returnMarcco replied:
This is now resolved

aNebula commented on src/app/projects/states/tutorials/tutorials.component.ts:
```diff
@@ -0,0 +1,162 @@
+
+@Component({
+  selector: 'tutorials',
+  templateUrl: './tutorials.component.tpl.html',
```
Please change html file name to take out `.tpl`

aNebula reviewed (changes requested):
Change HTML filename please

returnMarcco committed:
chore: amend file name

- change template file name to adhere to styling convention
- add `todo` comment to Typescript file

returnMarcco commented:
@aNebula Thanks for the feedback. I've completed the requested changes. I've noticed that my tutorial objects no longer have a defined `tutor_name` property on them. As can be seen in the screenshot above, this was defined when I finished the migration and completed testing. I am in the `doubtfire-api:9.x` branch for the backend. Could there be possible backend changes that have affected this field?

aNebula commented:
> @aNebula Thanks for the feedback. I've completed the requested changes. I've noticed that my tutorial objects no longer have a defined `tutor_name` property on them. As can be seen in the screenshot above, this was defined when I finished the migration and completed testing. I am in the `doubtfire-api:9.x` branch for the backend. Could there be possible backend changes that have affected this field?

That shouldn't be the case. I took a look at the https://github.com/thoth-tech/doubtfire-api/blob/9.x/app/api/tutorials_api.rb - it hasn't changed in the last 3 years. You might have to look into whether anything changed on the FE service or any issues during your testing.

returnMarcco commented:
> > @aNebula Thanks for the feedback. I've completed the requested changes. I've noticed that my tutorial objects no longer have a defined `tutor_name` property on them. As can be seen in the screenshot above, this was defined when I finished the migration and completed testing. I am in the `doubtfire-api:9.x` branch for the backend. Could there be possible backend changes that have affected this field?
> 
> That shouldn't be the case. I took a look at the https://github.com/thoth-tech/doubtfire-api/blob/9.x/app/api/tutorials_api.rb - it hasn't changed in the last 3 years. You might have to look into whether anything changed on the FE service or any issues during your testing.

I've tested this in the base `9.x` branch and the same issue occurs. It doesn't seem to be a result of my changes. I'm going to raise an upstream PR as the ticket is in the `Sprint 4 Upstream Review` bucket in Planner.

returnMarcco committed:
Migrate/group set selector (#981)

* chore: migrate group-set-selector (WIP)

- define and declare new component files

- import new component into app

- downgrade component

* chore: migrate group-set-selector

- convert AngularJS template syntax to Angular syntax
- finish implementing `group-set-selector.component.ts`
- remove redundant Coffeescript file

* chore: migrate group-set-selector

- finish defining Typescript logic
- update parent component to use Angular syntax for child component declaration

* chore: cleanup and documentation

- remove redundant Coffeescript file
- adjust frontend migration todo list

* chore: remove redundant file

- remove redundant Coffeescript file

* fix: attribute and event binding

- fix attribute and event binding syntax
- remove redundant template file

* refactor: two-way binding

- remove redundant directive attributes

- rename EventEmitter var name in child component

* Update README.md

* Update README.md

* refactor: replace Bootstrap components with Angular Material

- replace Bootstrap components with Angular Material components & directives

- refactor to use modern Angular structural directives

returnMarcco committed:
refactor: migrate/date service (#931)

* chore: date-service.coffee to TS

- add `date.service.ts` file
- add `date.service.spec.ts` unit test
- downgrade TS module such that it works within AngularJS

* chore: unlink date-service.coffee

- complete unlinking process of  `date-service.coffee` after migrating to Typescript
- manual linting/comment formatting

* chore: delete redundant coffeescript file

- delete `date-service.coffee` file after migration

* chore:  formatting

- format code with prettier

lachlan-robinson committed:
refactor: migrate progress dashboard (#927)

* chore: 9.x windows

* chore: create new component files

* chore: unlink old component + link new component

* chore: migrate functionality + update to tailwindcss and material UI

* chore: remove old component files

* chore: responsive columns for visualisations

* Update package-lock.json

* Update package.json

* Update package.json

* Update package-lock.json

* chore: update styling to better match original

* chore: add custom breakpoint for responsive change

* chore: tweak breakpoint for better visualisation visibility

* chore: update tutor view functionality + tweak event emitter name

---------

Co-authored-by: Boink <40929320+b0ink@users.noreply.github.com>

lachlan-robinson committed:
fix: burndown chart visualisation (#942)

* fix: update burndown data to render series of different lengths

* chore: cleanup imports

* chore: remove leftover testing comment from previous contributor

b0ink committed:
Merge pull request #929 from zara1504/remove/media-service

Remove Media Service

lachlan-robinson committed:
refactor: migrate task status pie chart (#928)

* chore: migrate pie chart to ngxchart

* chore: remove testing console log

* chore: restore package and lock to 9.x

* chore: remove testing comments

* chore: resolve merge conflicts

* chore: remove testing console log

* chore: resolve futher conflicts

* chore: exlude packge-lock

* chore: exclude package

* chore: delete migrated progress dashboard html

---------

Co-authored-by: Boink <40929320+b0ink@users.noreply.github.com>

returnMarcco committed:
refactor: migrate/unit staff editor 9.x (#933)

* chore: migrate/unit-staff-editor

- delete old Coffeescript file
- delete old template
- remove reference to old Coffeescript file
- link new component
- downgrade new component

* chore: migrate/unit-staff-editor

- port `unit-staff-editor-component` to Typescript from Coffeescript
- port `unit-staff-editor` template to Angular template syntax w/ Angular Material
- adjust parent component `Inputs` and attribute bindings

* Update README.md

* Update README.md

---------

Co-authored-by: Boink <40929320+b0ink@users.noreply.github.com>

b0ink committed:
refactor: unit staff editor material ui (#1001)

* fix: revert staff role on error

* fix: ensure full search for unit role

* chore: fix casing

* refactor: use mat table for staff editor

* chore: remove unused variables

* fix: null check

* refactor: use icon button

* chore: remove unit staff editor coffeescript file

* chore: add back staff editor tooltips

b0ink committed:
fix: check for valid unit

b0ink committed:
Merge branch 'migrate/tutorials' into migrate/tutorials
